src/client/dfs/dfs.c

```diff
@@ -1883,6 +1883,14 @@ dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj,
 				D_GOTO(err_obj, rc = ENOENT);
 			}
 
+			rc = check_access(dfs, geteuid(), getegid(), entry.mode,
```

Comment:
This checks access for S_ISREG, but do we also need to check for directories as well? Below, on line 1963:
```
/* open the directory object */
rc = daos_obj_open(dfs->coh, entry.oid, daos_mode, &obj->oh,
		NULL);
```
Do we need to check there as well? At the beginning of the loop, we check parent directory access, but if the leaf happens to be a directory, we don't go to the beginning of the loop, so I would think this directory would not be checked.
Also, do we need cleanup for a failed check_access? I also wonder if that first check_access for the parent also needs cleanup, since `obj` has been allocated
`D_GOTO(err_obj, rc);`

Actually, it also seems that `rem` isn't always free'ed for `D_GOTO(err_obj, ...)`
Maybe I'm missing something?

Reply:
yea you're right, for the dir part i was thinking the X_OK check was sufficient, but we need the R/W check on the last one.